packages/foam-vscode/src/features/tags-tree-view/index.ts

```diff
@@ -63,52 +63,37 @@ export class TagsProvider implements vscode.TreeDataProvider<TagTreeItem> {
   }
 
   getTreeItem(element: TagTreeItem): vscode.TreeItem {
-    if (element.title.indexOf(TAG_SEPARATOR) > -1) {
-      return;
-    }
     return element;
   }
 
   getChildren(element?: Tag): Thenable<TagTreeItem[]> {
     if (element) {
-      const tagChilds: TagTreeItem[] = this.tags
-        .filter(tag => tag.tag.indexOf(element.tag + TAG_SEPARATOR) > -1)
+      const nestedTagItems: TagTreeItem[] = this.tags
+        .filter(item => item.tag.indexOf(element.tag + TAG_SEPARATOR) > -1)
         .map(
-          ({ tag, notes }) =>
-            new Tag(tag, tag.substring(tag.indexOf(TAG_SEPARATOR) + 1), notes)
-        );
+          item =>
+            new Tag(
+              item.tag,
+              item.tag.substring(item.tag.indexOf(TAG_SEPARATOR) + 1),
+              item.notes
+            )
+        )
+        .sort((a, b) => a.title.localeCompare(b.title));
 
       const references: TagTreeItem[] = element.notes
         .map(({ uri }) => this.foam.workspace.get(uri))
-        .map(note => new TagReference(element.tag, note) as TagTreeItem)
-        .concat(tagChilds);
+        .map(note => new TagReference(element.tag, note))
+        .sort((a, b) => a.title.localeCompare(b.title));
 
       return Promise.resolve([
         new TagSearch(element.tag),
-        ...references
-          .sort((a, b) => {
-            if (a instanceof Tag && b instanceof TagReference) {
-              return -1;
-            } else {
-              return 1;
-            }
-          })
-          .sort((a, b) => {
-            if (
-              (a instanceof Tag && b instanceof Tag) ||
-              (a instanceof TagReference && b instanceof TagReference)
-            ) {
-              a.title.localeCompare(b.title);
-            }
-
-            return 0;
-          }),
+        ...nestedTagItems.concat(references),
```

Comment:
very minor: to be in line with the general Foam (unformalized) code style, could you change this to

```suggestion
        ...nestedTagItems,
        ...references,
```